modules/platform/nuxeo-platform-filemanager-core/src/main/java/org/nuxeo/ecm/platform/filemanager/service/extension/FileImporterDescriptor.java

```diff
@@ -173,23 +176,14 @@ public Descriptor merge(Descriptor o) {
 
         FileImporterDescriptor merged = new FileImporterDescriptor();
         merged.name = other.name;
-        merged.enabled = other.enabled;
-        merged.klass = defaultValue(other.klass, klass);
-        merged.className = defaultValue(other.className, className);
-        merged.docType = defaultValue(other.docType, docType);
+        merged.enabled = defaultIfNull(other.enabled, enabled);
+        merged.klass = defaultIfNull(other.klass, klass);
+        merged.className = defaultIfBlank(other.className, className);
+        merged.docType = defaultIfBlank(other.docType, docType);
         merged.filters = new ArrayList<>();
         merged.filters.addAll(filters);
         merged.filters.addAll(other.filters);
-        merged.order = defaultValue(other.order, order);
+        merged.order = defaultIfNull(other.order, order);
         return merged;
     }
-
-    protected <T> T defaultValue(T value, T defaultValue) {
-        return value == null ? defaultValue : value;
-    }
-
-    @Override
-    public boolean doesRemove() {
-        return !enabled;
```

Comment:
Why can it be safely removed?

Reply:
doesRemove makes the last descriptor null which is filtered out by the registry. Now that the registry takes care to filter out disabled contributions, this override is not needed anymore.

Furthermore, the doesRemove is better suited to re-init the contributions merge with such contributions:

```
<extension target="" point="">
  <something remove="true" />
  <something ...> <!-- completely new contribution -->
     ...
  </something>
</extension>
```

Whereas enabled with help of merge attribute allows us to do what we want in a single contribution.